### PR TITLE
feat(app): update ODD device reset page

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

This PR updates the robot settings/device reset page on the ODD to: 

1. remove clearing custom bootscripts as an option
2. add a`clear all stored data` option
3. add a confirmation modal before resetting and restarting the robot
4. add a robot side analytics setting. this will be updated in a follow up since there are two analytics settings (robot side and app side)

closes RAUT-639

# Test Plan
Verified functionality on a Flex dev kit

# Changelog

- Remove clearing custom bootscripts as an option
- Add a clear all stored data option
- Add a confirmation modal before resetting and restarting the robot

# Risk assessment

Low

### CODE REVIEW

File: app/src/pages/OnDeviceDisplay/RobotSettingsDashboard/RobotSettingsList.tsx [hunk 1]
```diff
@@ -65,28 +74,21 @@ export function RobotSettingsList(props: RobotSettingsListProps): JSX.Element {
   const devToolsOn = useSelector(getDevtoolsEnabled)
   const historicOffsetsOn = useSelector(getApplyHistoricOffsets)
   const { lightsEnabled, toggleLights } = useLEDLights(robotName)
-  const settings = useSelector<State, RobotSettings>((state: State) =>
-    getRobotSettings(state, robotName)
-  )
-  const homeGantrySettings = settings?.find(s => s.id === 'disableHomeOnBoot')
-
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/RobotSettingsDashboard/RobotSettingsList.tsx:72]: Did we scope out `Home Gantry On Restart`?

Comment (shlokamin) [reply app/src/pages/OnDeviceDisplay/RobotSettingsDashboard/RobotSettingsList.tsx:72]: it was a mistake — i added it back. thank you!

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 2]
```diff
@@ -195,6 +200,47 @@ export function DeviceReset({
               </React.Fragment>
             )
           })}
+
+          <OptionButton
+            id={'clearAllStoredData'}
+            type="checkbox"
+            value={'clearAllStoredData'}
```
Comment (jerader) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]:
```suggestion
            id="clearAllStoredData"
            type="checkbox"
            value="clearAllStoredData"
```

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 3]
```diff
@@ -195,6 +200,47 @@ export function DeviceReset({
...
+                          [val.id]: true,
+                        }
+                      },
+                      { clearAllStoredData: true }
+                    )
+              )
+            }}
+          />
+          <OptionLabel
+            htmlFor={'clearAllStoredData'}
```
Comment (jerader) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx]:
```suggestion
            htmlFor="clearAllStoredData"
```

File: app/src/pages/OnDeviceDisplay/RobotDashboard/WelcomeModal.tsx [hunk 4]
```diff
@@ -24,7 +24,7 @@ interface WelcomeModalProps {
   setShowWelcomeModal: (showWelcomeModal: boolean) => void
 }
 
-export function WelcomedModal({
+export function WelcomeModal({
```
Comment (jerader) [inline app/src/pages/OnDeviceDisplay/RobotDashboard/WelcomeModal.tsx:27]: womp good catch

File: app/src/pages/OnDeviceDisplay/RobotSettingsDashboard/RobotSettingsList.tsx [hunk 5]
```diff
@@ -29,33 +28,51 @@ import {
...
 import { useNetworkConnection } from '../hooks'
-import { getRobotSettings } from '../../../redux/robot-settings'
 import { RobotSettingButton } from './RobotSettingButton'
 
 import type { Dispatch, State } from '../../../redux/types'
-import type { RobotSettings } from '../../../redux/robot-settings/types'
 import type { SetSettingOption } from './'
+import { StyledText } from '../../../atoms/text'
+import { InlineNotification } from '../../../atoms/InlineNotification'
+import { getRobotSettings, updateSetting } from '../../../redux/robot-settings'
```
Comment (jerader) [inline app/src/pages/OnDeviceDisplay/RobotSettingsDashboard/RobotSettingsList.tsx]: these imports can go higher

File: app/src/pages/OnDeviceDisplay/RobotSettingsDashboard/__tests__/RobotSettingsDashboard.test.tsx [hunk 6]
```diff
@@ -140,7 +140,8 @@ describe('RobotSettingsDashboard', () => {
     getByText('Use stored data when setting up a protocol.')
     getByText('Developer Tools')
     getByText('Access additional logging and feature flags.')
-    expect(getAllByText('Off').length).toBe(3) // LED & DEV tools & historic offsets
+    getByText('Share Analytics')
+    expect(getAllByText('Off').length).toBe(5) // LED & DEV tools & share analytics & home gantry & historic offsets
   })
 
   // Note(kj: 02/03/2023) This case will be changed in a following PR
```
Comment (jerader) [inline app/src/pages/OnDeviceDisplay/RobotSettingsDashboard/__tests__/RobotSettingsDashboard.test.tsx]: is this note still needed?

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx [hunk 7]
```diff
@@ -195,6 +200,47 @@ export function DeviceReset({
...
+            isSelected={resetOptions.clearAllStoredData}
+          >
+            <Flex flexDirection={DIRECTION_COLUMN}>
+              <StyledText as="p" fontWeight={TYPOGRAPHY.fontWeightSemiBold}>
+                {t('clear_all_stored_data')}
+              </StyledText>
+              <StyledText
+                as="p"
+                color={
+                  resetOptions.clearAllStoredData === true
```
Comment (jerader) [inline app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx:235]: does this work? 
```suggestion
                  resetOptions.clearAllStoredData
```

**Review by jerader — APPROVED** [review]
i left a handful of cleanup comments but lgtm

File: app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx (hunk 7 above)
Comment (shlokamin) [reply app/src/organisms/RobotSettingsDashboard/DeviceReset.tsx:235]: yeah but itll give us an es-lint warning cuz the type of `resetOptions.clearAllStoredData` is `boolean | undefined`, if it were just a `boolean` this wouldnt give us a warning though!

**Pull request merged**